### PR TITLE
v23.2.4 release notes

### PR DESCRIPTION
**Pull request by florence-crl**
Fixes DOC-9904

v23.2.4 release notes edited and cloud current version note removed
sha updated

### CODE REVIEW

File: src/current/_data/releases.yml [hunk 1]
```diff
@@ -5762,3 +5762,37 @@
     docker_arm_limited_access: false
   source: true
   previous_release: v24.1.0-alpha.4
+
+- release_name: v23.2.4
+  major_version: v23.2
+  release_date: '2024-04-08'
```
Comment (taroface) [inline src/current/_data/releases.yml:5768]: Just confirming this is the correct date?

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+## v23.2.4
+
+Release Date: April 8, 2024
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:3]: Ditto

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-2-4-sql-language-changes">SQL language changes</h3>
+
+- Mutation statements such as [`UPDATE`](../v23.2/update.html) and [`DELETE`](../v23.2/delete.html) as well as locking statements such as [`SELECT FOR UPDATE`](../v23.2/select-for-update.html) are not allowed in [read-only transactions](../v23.2/set-transaction.html#parameters) or [`AS OF SYSTEM TIME` transactions](../v23.2/set-transaction.html#parameters). Fixed a bug where mutation statements and locking statements in implicit single-statement transactions using AS OF SYSTEM TIME were allowed. [#120158][#120158]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:9]:
```suggestion
- Mutation statements such as [`UPDATE`](../v23.2/update.html) and [`DELETE`](../v23.2/delete.html) as well as locking statements such as [`SELECT FOR UPDATE`](../v23.2/select-for-update.html) are not allowed in [read-only transactions](../v23.2/set-transaction.html#parameters) or [`AS OF SYSTEM TIME` transactions](../v23.2/set-transaction.html#parameters). This is the result of fixing a bug where mutation statements and locking statements in implicit single-statement transactions using `AS OF SYSTEM TIME` were allowed. [#120158][#120158]
```

Comment (taroface) [reply src/current/_includes/releases/v23.2/v23.2.4.md:9]: This is kind of a wordy suggestion, but I feel like the relationship between the two sentences wasn't clear. Feel free to adjust further.

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- The new cluster setting [`sql.stats.virtual_computed_columns.enabled`](../v23.2/cluster-settings.html#setting-sql-stats-virtual-computed-columns-enabled), when enabled, allows the collection of [table statistics](../v23.2/show-statistics.html) on [virtual computed columns](../v23.2/computed-columns.html). [#120933][#120933]
+- The new [session variable](../v23.2/session-variables.html) `optimizer_use_virtual_computed_column_stats`, when enabled, configures the [cost-based optimizer](../v23.2/cost-based-optimizer.html) to use [table statistics](../v23.2/show-statistics.html) on [virtual computed columns](../v23.2/computed-columns.html). [#121329]
+
+<h3 id="v23-2-4-db-console-changes">DB Console changes</h3>
+
+- Fixed an issue where clusters with multiple [stores](../v23.2/cockroach-start.html#store) per node may list inaccurate region and node information on the [Databases page](../v23.2/ui-databases-page.html#databases). [#120212][#120212]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:15]:
```suggestion
- Fixed an issue where clusters with multiple [stores](../v23.2/cockroach-start.html#store) per node could list inaccurate region and node information on the [**Databases** page](../v23.2/ui-databases-page.html#databases). [#120212][#120212]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- Users will no longer see [views](../v23.2/views.html) displayed on the [Databases page](../v23.2/ui-databases-page.html#databases). Previously views would be listed with no information, only displaying errors. [#120214][#120214]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:16]:
```suggestion
- Users will no longer see [views](../v23.2/views.html) displayed on the [**Databases** page](../v23.2/ui-databases-page.html#databases). Previously views would be listed with no information, only displaying errors. [#120214][#120214]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+<h3 id="v23-2-4-bug-fixes">Bug fixes</h3>
+
+- Previously, [altering a table's locality](../v23.2/alter-table.html#set-locality) from [`REGIONAL BY ROW`](../v23.2/alter-table.html#set-the-table-locality-to-regional-by-row) to [`REGIONAL BY TABLE`](../v23.2/alter-table.html#regional-by-table) could cause [leaseholders](../v23.2/architecture/replication-layer.html#leases) to never move to the [database's primary region](../v23.2/alter-database.html#set-primary-region). This is now fixed. [#118794][#118794]
+- A user with the [`VIEWACTIVITYREDACTED` privilege](../v23.2/security-reference/authorization.html#privileges) should not be able to see constants inside of queries that originate from other users in the [`SHOW SESSIONS` result](../v23.2/show-sessions.html#response). Previously, this redaction did not occur, but this is fixed now. [#119884][#119884]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:21]:
```suggestion
- A user with the [`VIEWACTIVITYREDACTED` privilege](../v23.2/security-reference/authorization.html#privileges) can no longer see constants inside of queries that originate from other users in the [`SHOW SESSIONS` result](../v23.2/show-sessions.html#response). Previously, this redaction did not occur. [#119884][#119884]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+- Previously, the [`SHOW STATEMENTS`](../v23.2/show-statements.html) and the [`SHOW QUERIES`](../v23.2/show-statements.html#aliases) commands were incorrectly requiring that the user has the [`VIEWACTIVITY` or `VIEWACTIVITYREDACTED` privilege](../v23.2/security-reference/authorization.html#privileges). However, a user always should be able to view their own queries, even without these privileges. This is fixed now. [#119884][#119884]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:22]:
```suggestion
- Previously, the [`SHOW STATEMENTS`](../v23.2/show-statements.html) and the [`SHOW QUERIES`](../v23.2/show-statements.html#aliases) commands incorrectly required the user to have the [`VIEWACTIVITY` or `VIEWACTIVITYREDACTED` privilege](../v23.2/security-reference/authorization.html#privileges). However, a user always should be able to view their own queries, even without these privileges. This is now fixed. [#119884][#119884]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+- Fixed a bug where [`RESTORE`](../v23.2/restore.html) on certain [`backups`](../v23.2/backup.html) would open a very large number of connections to the backup storage provider. [#119883][#119883]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:23]:
```suggestion
- Fixed a bug where [`RESTORE`](../v23.2/restore.html) on certain [`BACKUP`s](../v23.2/backup.html) would open a very large number of connections to the backup storage provider. [#119883][#119883]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+- Fixed a bug that occurred when using [`ALTER TABLE`](../v23.2/alter-table.html) to [drop](../v23.2/alter-table.html#drop-constraint) and [add](../v23.2/alter-table.html#add-constraint) back a [CHECK constraint](../v23.2/check.html) with the same name. [#120075][#120075]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:24]:
```suggestion
- Fixed a bug that occurred when using [`ALTER TABLE`](../v23.2/alter-table.html) to [drop](../v23.2/alter-table.html#drop-constraint) and [add](../v23.2/alter-table.html#add-constraint) back a [`CHECK` constraint](../v23.2/check.html) with the same name. [#120075][#120075]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+- Fixed a bug in which it was possible to set [session variable](../v23.2/session-variables.html) `transaction_read_only` to `false` during an [AS OF SYSTEM TIME transaction](../v23.2/set-transaction.html#parameters). [#120158][#120158]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:25]:
```suggestion
- Fixed a bug in which it was possible to set the [session variable](../v23.2/session-variables.html) `transaction_read_only` to `false` during an [`AS OF SYSTEM TIME` transaction](../v23.2/set-transaction.html#parameters). [#120158][#120158]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+- Fixed a bug where some files were not closed when inspecting [backup metadata](../v23.2/backup-architecture.html#metadata-writing-phase) during [BACKUP](../v23.2/backup.html) and [RESTORE](../v23.2/restore.html). [#119635][#119635]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:26]:
```suggestion
- Fixed a bug where some files were not closed when inspecting [backup metadata](../v23.2/backup-architecture.html#metadata-writing-phase) during [`BACKUP`](../v23.2/backup.html) and [`RESTORE`](../v23.2/restore.html). [#119635][#119635]
```

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+- Fixed an intermittent page crash on the [Schema Insights](../cockroachcloud/insights-page.html#schema-insights-tab) view. [#120210][#120210]
```
Comment (taroface) [inline src/current/_includes/releases/v23.2/v23.2.4.md:27]:
```suggestion
- Fixed an intermittent page crash on the [**Schema Insights**](../cockroachcloud/insights-page.html#schema-insights-tab) view. [#120210][#120210]
```

**Review by taroface — APPROVED** [review]
LGTM, just various nits

File: src/current/_data/releases.yml (hunk 1 above)
Comment (florence-crl) [reply src/current/_data/releases.yml:5768]: good catch! updated to 04-09

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 2 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:3]: good catch! updated to 04-09

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 3 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:9]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 4 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:15]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 5 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:16]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 6 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:21]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 7 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:22]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 9 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:24]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 10 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:25]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 11 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:26]: updated

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 12 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md:27]: updated

**Review by florence-crl — COMMENTED** [review]
TFTR!

File: src/current/_includes/releases/v23.2/v23.2.4.md (hunk 8 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.2/v23.2.4.md]: updated

**Comment by florence-crl** [comment]
Hi @taroface , I just reassigned you to review because the sha was changed ([REL-745](https://cockroachlabs.atlassian.net/browse/REL-745?focusedCommentId=219941)). No new release notes were added.

**Review by taroface — APPROVED** [review]

**Comment by florence-crl** [comment]
Hi @mikeCRL, I assigned you to review not the release notes, but the removal of the cloud variables. thanks!

**Review by mikeCRL — APPROVED** [review]

**Comment by florence-crl** [comment]
Hi @taroface and @yuzefovich,  please review the [release note](https://deploy-preview-18441--cockroachdb-docs.netlify.app/docs/releases/v23.2.html##v23-2-4-bug-fixes) for 1 bugfix ([121875.](https://github.com/cockroachdb/docs/pull/18441/commits/d751d906a449f88bd0cd137a4f6f845e5dbb47f1)) from the last sha change.

**Review by taroface — APPROVED** [review]
LGTM!

File: src/current/_includes/releases/v23.2/v23.2.4.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+<h3 id="v23-2-4-db-console-changes">DB Console changes</h3>
+
+- Fixed an issue where clusters with multiple [stores](../v23.2/cockroach-start.html#store) per node could list inaccurate region and node information on the [**Databases** page](../v23.2/ui-databases-page.html#databases). [#120212][#120212]
+- Users will no longer see [views](../v23.2/views.html) displayed on the [**Databases** page](../v23.2/ui-databases-page.html#databases). Previously views would be listed with no information, only displaying errors. [#120214][#120214]
+
+<h3 id="v23-2-4-bug-fixes">Bug fixes</h3>
+
+- Previously, on long-running [sessions](../v23.2/show-sessions.html) that issue many (hundreds of thousands or more) [transactions](../v23.2/transactions.html), CockroachDB's internal memory accounting system, the limit for which is configured via the [`--max-sql-memory` flag](../v23.2/cockroach-start.html#general)) could leak. This bug, in turn, could result in the error message `"root: memory budget exceeded"` for other queries. The bug was present in v23.1.17 and is now fixed. [#121875][#121875]
```
Comment (yuzefovich) [inline src/current/_includes/releases/v23.2/v23.2.4.md:20]: The bug is also present on 23.2.3 version (i.e. only specifically 23.1.17 and 23.2.3). (I did mention this in the original release note to https://github.com/cockroachdb/cockroach/pull/121875, but I think the mention of 23.2.3 version was consciously omitted from 23.1.18 release notes.)

**Review by yuzefovich — APPROVED** [review]

**Pull request merged**